src/app/features/project/metadata/project-metadata.component.ts

```diff
@@ -1,19 +1,577 @@
+        type: 'cedar' as const,
+      }));
+
+      this.tabs.set([...baseTabs, ...cedarTabs]);
+
+      this.handleRouteBasedTabSelection();
+    });
+
+    effect(() => {
+      const templates = this.cedarTemplates();
+      const selectedRecord = this.selectedCedarRecord();
+
+      if (selectedRecord && templates?.data && !this.selectedCedarTemplate()) {
+        const templateId = selectedRecord.relationships?.template?.data?.id;
+        if (templateId) {
+          const template = templates.data.find((t) => t.id === templateId);
+          if (template) {
+            this.selectedCedarTemplate.set(template);
+          }
+        }
+      }
+    });
+  }
+
+  ngOnInit(): void {
+    const projectId = this.route.parent?.parent?.snapshot.params['id'];
+
+    if (projectId) {
+      this.actions.getProject(projectId);
+      this.actions.getCustomItemMetadata(projectId);
+      this.actions.getContributors(projectId);
+      this.actions.getHighlightedSubjects();
+      this.actions.getCedarRecords(projectId);
+      this.actions.getCedarTemplates();
+
+      const user = this.currentUser();
+      if (user?.id) {
+        this.actions.getUserInstitutions(user.id);
+      }
+    }
+
+    this.route.paramMap.pipe(takeUntilDestroyed(this.destroyRef)).subscribe((params) => {
+      this.handleRouteBasedTabSelection();
+    });
+  }
+
+  private handleRouteBasedTabSelection(): void {
+    const recordId = this.route.snapshot.paramMap.get('recordId');
+
+    if (!recordId) {
+      this.activeTabIndex.set(0);
+      this.selectedCedarRecord.set(null);
+      this.selectedCedarTemplate.set(null);
+      return;
+    }
+
+    const tabs = this.tabs();
+    const index = tabs.findIndex((tab) => tab.id === recordId);
+
+    if (index >= 0) {
+      this.activeTabIndex.set(index);
+      const tab = tabs[index];
+      if (tab.type === 'cedar') {
+        this.loadCedarRecord(tab.id);
+      }
+    }
+  }
+
+  onTagsChanged(tags: string[]): void {
+    const projectId = this.currentProject()?.id;
+    if (projectId) {
+      this.actions.updateProjectDetails(projectId, { tags });
+    }
+  }
+
+  openAddRecord(): void {
+    this.router.navigate(['add'], { relativeTo: this.route });
+  }
+
+  onSubjectsChanged(subjects: ProjectOverviewSubject[]): void {
+    const projectId = this.currentProject()?.id;
+    if (projectId) {
+      const subjectIds = subjects.map((subject) => subject.id);
+      this.actions.updateProjectSubjects(projectId, subjectIds);
+    }
+  }
+
+  openEditContributorDialog(): void {
+    const dialogRef = this.dialogService.open(ContributorsDialogComponent, {
+      header: this.translateService.instant('project.metadata.contributors.dialog.header'),
+      width: '800px',
+      data: {
+        projectId: this.currentProject()?.id,
+        contributors: this.contributors(),
+        isLoading: this.isContributorsLoading(),
+      },
+    });
+
+    dialogRef.onClose.pipe(filter((result) => !!result)).subscribe({
```

Comment:
You can do lke this:
```suggestion
    dialogRef.onClose.pipe(filter((result) => !!result && (result.refresh || result.saved))).subscribe({
```
It will remove `if`